### DISCUSSION TITLE
[Docs] SwitchMode vs ButtonTopic 1 total confusion

### DISCUSSION THREAD

**Discussion by goblin** [arendst/Tasmota#19195] [Q&A]
Hello,

Where can I report bugs relating to the documentation of Tasmota (specifically https://tasmota.github.io/docs/Buttons-and-Switches/)? When opening an Issue, it says it's only meant for software problems.

### Issue 1

The first issue I have is that first (at https://tasmota.github.io/docs/Buttons-and-Switches/#switchmode) I read that:

> SwitchMode has **NO** impact on the behavior of GPIO configured as Button<x> [components](https://tasmota.github.io/docs/Components/).

But later on (at https://tasmota.github.io/docs/Buttons-and-Switches/#buttontopic) I read that for `ButtonTopic 1`:

> When changing the state of the button an MQTT message is sent to the device topic with the payload according to `SwitchMode` set.

I suspect that ButtonTopic is only used for... well, Buttons, not Switches. And if it does something according to `SwitchMode`, then it's not true that "SwitchMode has **NO** impact" on Buttons.

### Issue 2

I currently have a momentary pushbutton from https://www.tasmota.shop/gb/switches/17-key-button-switch-3x-tasmota.html configured (from factory) as a Button. In https://www.tasmota.shop/gb/switches/17-key-button-switch-3x-tasmota.html we read that "1 short press" **toggles** the power state by sending a MQTT status message like `stat/tasmota/POWER = ON`. And indeed it seems to be doing that in my (modified) setup (with `SetOption73` set to `ON`, among possibly other things). **However**,  `POWER = ON` **switches the power on**, not **toggles** it! It manifests itself in that when I press a button when power is off, the relay gets turned on, and when I press it again, it stays on. How can I change it so that `stat/tasmota/POWER1 = TOGGLE` message is sent instead (and why do docs say "toggle" followed by "on")?

Overall, I'm very confused by this whole thing. There's a ton of moving parts and lots of weird options and exceptions. I'd much prefer to just have a factory rule that changes the corresponding relay whenever a button is pressed, with no special options, topics and all that. Then anyone could just change that rule to additionally send an MQTT message or do whatever they wanted.

**Comment by goblin** [comment]
Furthermore,

### Issue 3

At https://tasmota.github.io/docs/Buttons-and-Switches/#buttontopic, in the Example for `ButtonTopic <value>`, we read:

> For example, we set the topic to *tasmota02* with `ButtonTopic tasmota02`. With `SwitchMode 1` the device yields the following message: `MQT: cmnd/tasmota02/POWER = TOGGLE`

However, I've done the exact same thing (set `ButtonTopic` to `tasmota02` and set `SwitchMode 1`), but I'm seeing `cmnd/tasmota02/POWER1 ON` instead of `TOGGLE`.

Also, still not sure why `SwitchMode 1` is mentioned there at all, while the docs for SwitchMode say it only affects Switches, not Buttons (see my Issue 1 above).

**Comment by Oekel** [comment]
Same to me. I am very happy about the possibilities tasmota brings tu us.
But I do not unterstand why we nee to decice for Button/Switch on the GPIO-Level when there is such a bunch of manipulations inside SwitchMode.

I thinks there might be a good reason (facing the fact that it has not been simplified until version 14)
But PLEASE let us understand it in detail and help each other to fix this documentation!

**Reply by arendst** [reply]
The text regarding buttons AND switchmode is wrong. There is no relation between buttons and switchmode. Switchmode ONLY applies to switches.

As the docs are open for anyone to edit I suppose someone using switches felt the need to bother the buttons text....
